### ISSUE TITLE
[QUEST] Usage of PSSM with new ProteinMPNN

### ISSUE THREAD

**Issue by SIAndersson** [RosettaCommons/foundry#332]
In a previous workflow of mine, I used PSSMs to bias sequence generation. I can't really find any documentation on how to use this. I see that there is a `PSSMMPNN `class defined in the repo, but from what I can tell, there is no straightforward way of giving it your PSSM file directly. Could this be "hacked" by using `bias_per_residue` in some way? And should one just use the regular ProteinMPNN in that case? Thanks in advance!

**Comment by AndrewKubaney** [comment]
Hello!

The `PSSMMPNN` class was mostly included as an example of how to subclass `ProteinMPNN` to add additional features / train model variants. It is not currently set up as a supported inference interface for passing in PSSM files.

For this use case, I think the best option is to use the regular `ProteinMPNN` model and pass the PSSM information through `bias_per_residue`.

The main caveat is that this bias is added to the amino acid logits before sampling. Roughly,

```
probs = softmax((logits + bias) / temperature)
```

The values passed to `bias_per_residue` should be in log/score-like units, rather than raw frequencies. Positive values will upweight an amino acid at that position, and negative values will downweight it. If your PSSM is already in log / score units, then you can try to use those values directly. You may still need to multiply the PSSM by an overall scale factor, since the magnitude of the PSSM scores may not match the magnitude of the `ProteinMPNN` logits.

**Comment by SIAndersson** [comment]
Ah, as I expected then. Thank you for the quick and informative reply! I will play around a bit with the scaling of the PSSM scores and see what works best.
